go-controller/pkg/util/ovs.go

```diff
@@ -235,34 +296,18 @@ func GetExec() kexec.Interface {
 var runCounter uint64
 
 func runCmd(cmd kexec.Cmd, cmdPath string, args ...string) (*bytes.Buffer, *bytes.Buffer, error) {
-	stdout := &bytes.Buffer{}
-	stderr := &bytes.Buffer{}
-
-	cmd.SetStdout(stdout)
-	cmd.SetStderr(stderr)
-
-	counter := atomic.AddUint64(&runCounter, 1)
-	logCmd := fmt.Sprintf("%s %s", cmdPath, strings.Join(args, " "))
-	klog.V(5).Infof("exec(%d): %s", counter, logCmd)
-
-	err := cmd.Run()
-	klog.V(5).Infof("exec(%d): stdout: %q", counter, stdout)
-	klog.V(5).Infof("exec(%d): stderr: %q", counter, stderr)
-	if err != nil {
-		klog.V(5).Infof("exec(%d): err: %v", counter, err)
-	}
-	return stdout, stderr, err
+	stdout, stderr, err := GetExecUtilRunSvc().RunCmd(cmd, cmdPath, []string{}, args...)
+	return bytes.NewBufferString(stdout), bytes.NewBufferString(stderr), err
 }
 
 func run(cmdPath string, args ...string) (*bytes.Buffer, *bytes.Buffer, error) {
-	cmd := runner.exec.Command(cmdPath, args...)
-	return runCmd(cmd, cmdPath, args...)
+	stdout, stderr, err := GetExecUtilRunSvc().Run(cmdPath, args...)
+	return bytes.NewBufferString(stdout), bytes.NewBufferString(stderr), err
```

Comment:
For the non-test path, why do we need to call `GetExecUtilRunSvc()`? Can't we just initialize a struct in ovs.go ahead of time and just use that. For example: In ovs.go,

var execRunner  ExecUtilRunSvc = &ExecUtilRunSvcImplStruct{}
and just use `execRunner` in run, runCmd, runWithEnvVars().

You can move GetExecUtilRunSvc() to ovs_unit_test.go

Reply:
Let me think through it,  try it out and provide an update

Reply:
the suggestion has been implemented